x-pack/plugins/fleet/server/services/epm/packages/install.ts

```diff
@@ -102,7 +102,11 @@ export async function ensureInstalledPackage(options: {
               pkgVersion: pkgKeyProps.version,
             },
           });
-    throw new Error(`${errorPrefix}: ${installResult.error.message}`);
+    const e = {
+      installType: installResult.installType,
+      error: new Error(`${errorPrefix}: ${installResult.error.message}`),
+    };
+    throw e;
```

Comment:
Can we return an object or throw an error? I was surprised to see throwing an object. I don't think we do this anywhere else

Reply:
I agree but it is feature freeze today and I've already had to refactor this PR like 3 or 4 times to reconcile with shifting requirements and other changes merging.

Reply:
The preconfiguration API only handles thrown errors, it has no system in place to handle a function call resolving with an error object instead of succeeding, so adding this right now would be another non-trivial change.

It's unfortunate that we didn't communicate about the default package error handling system while it was being worked on in parallel to this, but we're down to the wire and this functions and preserves all the additional metadata for us to iterate on in the future, so I really would prefer not to do another major change before merging.

Reply:
Lets remove pressure here, is there any strong problems to merge it like this? @jfsiii and @jen-huang  ?